commit 7f6240858cf3abb75237c9ba63ec70d232573ae8
Author: Srinivas Ramana <quic_sramana@quicinc.com>
Date:   Mon Feb 8 09:57:30 2021 +0000

    arm64: Defer enabling pointer authentication on boot core
    
    Defer enabling pointer authentication on boot core until
    after its required to be enabled by cpufeature framework.
    This will help in controlling the feature dynamically
    with a boot parameter.
    
    Signed-off-by: Ajay Patil <pajay@qti.qualcomm.com>
    Signed-off-by: Prasad Sodagudi <psodagud@codeaurora.org>
    Signed-off-by: Srinivas Ramana <sramana@codeaurora.org>
    Signed-off-by: Marc Zyngier <maz@kernel.org>
    Link: https://lore.kernel.org/r/1610152163-16554-2-git-send-email-sramana@codeaurora.org
    Reviewed-by: Catalin Marinas <catalin.marinas@arm.com>
    Acked-by: David Brazdil <dbrazdil@google.com>
    Link: https://lore.kernel.org/r/20210208095732.3267263-22-maz@kernel.org
    Signed-off-by: Will Deacon <will@kernel.org>

diff --git a/arch/arm64/kernel/head.S b/arch/arm64/kernel/head.S
index e7f12bf49730..1e30b5550d2a 100644
--- a/arch/arm64/kernel/head.S
+++ b/arch/arm64/kernel/head.S
@@ -404,10 +404,6 @@ SYM_FUNC_START_LOCAL(__primary_switched)
 	adr_l	x5, init_task
 	msr	sp_el0, x5			// Save thread_info
 
-#ifdef CONFIG_ARM64_PTR_AUTH
-	__ptrauth_keys_init_cpu	x5, x6, x7, x8
-#endif
-
 	adr_l	x8, vectors			// load VBAR_EL1 with virtual
 	msr	vbar_el1, x8			// vector table address
 	isb